azure-container-apps-mtls-certificate-filter/README.md

```diff
@@ -0,0 +1,134 @@
+- [Reactive Web App](azure-container-apps-mtls-certificate-filter-reactive): This module demonstrates how to extract certificates in a reactive web application.
+
+All the 3 modules have the similar structure, which includes a certificate filter and a validator.
+
+One main difference between the servlet modules and the reactive module is that the servlet ones add a [FilterRegistrationBean](azure-container-apps-mtls-certificate-filter-jakarta/src/main/java/com/microsoft/sample/security/servlet/FilterRegistrationBeanConfig.java) to register the certificate filter to the servlet container, whereas the reactive application can directly apply a web filter bean.
+
+## Main components
+
+As mentioned above, the main components of the project include an X.509 certificate filter and a certificate validator.
+
+### X.509 Certificate Filter
+
+The certificate filter is responsible for intercepting incoming requests, extracting the X.509 certificates from [Azure Container Apps' specific header](https://learn.microsoft.com/azure/container-apps/ingress-overview#http-headers), and storing them in the request attribute for further usages. Then the application can access the certificates from the attribute.
+
+### Certificate Validator
+
+The certificate validator is only an example, it should be implemented according to your own business requirement and the specific Java framework you are using. The validator checks the X.509 certificates extracted by the certificate filter, and it loads the server application's own trust store, which can contains the public and private CA certificates. So to use this validator, you should first prepare a trust store in your Container App and configure the trust store path and password in the application properties.
+
+## How to run
+
+To deploy this application to Azure Container Apps, you need to prepare the following resources:
+
+1. Prepare a trust store containing your CA certificates.
+2. Deploy your app to Azure Container Apps.
+3. Configure your container app to require client certificates.
+4. Test the application.
+
+The following sections provide detailed steps for each of the above operations and takes the [azure-container-apps-mtls-certificate-filter-jakarta](azure-container-apps-mtls-certificate-filter-jakarta) module as an example.
+
+### Prepare a trust store and certificates for testing
+
+In this section, we will prepare a trusted CA certificate store and the client certificates which can be verified by one of the CA certificate in the store. In production use case, the trusted CA certificates and the client certificates are likely to be provided by some authority. Here, for demonstration, we will use OpenSSL to generate the certificates, which should be used in development environment only. The below operations use the [openssl](https://wiki.openssl.org/index.php/Binaries) library, so your need to have it installed prior to following the next step.
```

Comment:
Let's make the sentence bold

Here, for demonstration, we will use OpenSSL to generate the certificates, which should be used in development environment only.